marketplace/commercelayer-products/src/index.js

```diff
@@ -0,0 +1,162 @@
+import CLayerAuth from '@commercelayer/js-auth';
+import CLayer from '@commercelayer/js-sdk';
+
+import { setup, renderSkuPicker } from 'shared-sku-app';
+
+import logo from './logo.svg';
+import descriptor from '../extension.json';
+import { dataTransformer, previewDataTransformer } from './dataTransformer';
+
+const PER_PAGE = 20;
+
+function makeCTA(fieldType) {
+  return fieldType === 'Array' ? 'Select products' : 'Select a product';
+}
+
+/**
+ * The Commerce Layer client is currently used to fetch only the previews
+ * for the selected products.
+ *
+ * To fetch all SKUs and to search through them we perform directly an HTTP
+ * request to the corresponding endpoint.
+ *
+ * @see fetchSKUs for a more detailed explanation.
+ */
+async function makeCommerceLayerClient({
+  parameters: {
+    installation: { apiEndpoint, clientId, clientSecret }
+  }
+}) {
+  const auth = await CLayerAuth.integration({
+    clientId,
+    clientSecret,
```

Comment:
OAuth client secrets should not be in the frontend. Do they not support an implicit flow?

Reply:
They do, shared-sku-picker doesn't :+1: We should probably revisit Commercetools then it's the same story there

Reply:
**Update**: now use the `salesChannel` flow as discussed which doesn't require a client secret

Reply:
awesome :+1:

Reply:
If we don't require it then why do we ask people to provide it in the configuration screen?